tests/t_signals.c

```diff
@@ -0,0 +1,97 @@
+#include <gdk-pixbuf/gdk-pixbuf.h>
+
+typedef struct
+{
+  gint area_prepared_cb_count;
+  gint area_updated_cb_count;
+  gint closed_cb_count;
+  gint size_prepared_cb_count;
+} SignalsCounters;
+
+void
+area_prepared_cb (GdkPixbufLoader* self, gpointer user_data)
+{
+  SignalsCounters *counters = (SignalsCounters *) user_data;
+
+  counters->area_prepared_cb_count++;
+}
+
+void
+area_updated_cb (GdkPixbufLoader* self, gint x, gint y, gint width, gint height, gpointer user_data)
+{
+  SignalsCounters *counters = (SignalsCounters *) user_data;
+
+  counters->area_updated_cb_count++;
+
+  g_assert_cmpint (x, ==, 0);
+  g_assert_cmpint (y, ==, 0);
+  g_assert_cmpint (width, ==, 200);
+  g_assert_cmpint (height, ==, 200);
+}
+
+void
+closed_cb (GdkPixbufLoader* self, gpointer user_data)
+{
+  SignalsCounters *counters = (SignalsCounters *) user_data;
+
+  counters->closed_cb_count++;
+}
+
+void
+size_prepared_cb (GdkPixbufLoader* self, gint width, gint height, gpointer user_data)
+{
+  SignalsCounters *counters = (SignalsCounters *) user_data;
+
+  counters->size_prepared_cb_count++;
+
+  g_assert_cmpint (width, ==, 200);
+  g_assert_cmpint (height, ==, 200);
+}
+
+gint
+main (gint argc, gchar **argv)
+{
+  gchar **env   = g_get_environ ();
+  GdkPixbufLoader *loader;
```

Comment:
Not a big fan of uninitialized pointers, please call gdk_pixbuf_loader_new() here or just declare the variable down below, we can use modern C patterns here.